### PR TITLE
Modernize library: native PSR-7, PHP 8.1+

### PR DESCRIPTION
**Pull request by gnutix**
## Summary

- Rewrite the library to use native PSR-7 `ResponseInterface` instead of the Slim-specific renderer, drop PHP 5.6-7.x support (require PHP 8.1+)
- Narrow `ContentType` interface return types (`mixed` → `?string`), add `JSON_THROW_ON_ERROR`, throw `InvalidArgumentException` on bad format in Text/Separated adapters
- Add full test suite (58 tests / 83 assertions), PHPStan max level (0 errors), ECS code style enforcement, modernized CI (PHP 8.1–8.5 matrix)

### Source (`lib/`)
- `ContentType` interface: return types narrowed from `mixed` to `?string`
- `Json`: `JSON_THROW_ON_ERROR` on both `json_encode` calls
- `Separated` / `Text`: throw `InvalidArgumentException` instead of returning error string on bad format
- `Separated` / `Csv` / `Tab`: `$separator` visibility `public` → `protected`
- `Renderer/Slim.php`: removed (replaced by native PSR-7 rendering)
- ECS code style applied across all files (imports, `?string` syntax, native function invocation)

### Tooling
- `phpunit.xml.dist`: new PHPUnit config with cache in `var/`
- `ecs.php`: ECS config (PSR-12 + Symfony sets, lean skip list)
- CI: PHP 8.1–8.5 test matrix, PHPStan + ECS jobs
- Removed `.phplint.yml` and `phpstan-baseline.neon`

### Tests
- `ContainerTest` (5 tests): factory/shared behavior, config defaults/merging
- `ProcessorTest` (22 tests): mapping, collections, dot-notation, sub-mapping, callables, format override
- `RendererTest` (27 tests): all content types, encoding, convert, error rendering, edge cases
- `ResponseTest` (4 tests): error helper, headers, ETag

## Test plan

- [x] `vendor/bin/phpunit` — 58 tests, 83 assertions pass
- [x] `vendor/bin/phpstan analyse` — 0 errors at max level
- [x] `vendor/bin/ecs check` — 0 violations
- [x] CI passes on all PHP versions (8.1–8.5)
- [ ] Verify integration with tipee

### CODE REVIEW

File: lib/ContentType/Standard.php [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
+        /** @var string|null */
         return $content;
     }
 
-    public function renderError($content, Renderer $renderer)
+    public function renderError(mixed $content, Renderer $renderer): ?string
     {
+        /** @var string|null */
         return $content;
     }
```
[the remark this answers is not in the document]
Comment (gnutix) [reply lib/ContentType/Standard.php:26]: Verified against tipee-cow: all `getRenderer()` calls use known mimetypes (`application/json`, `text/csv`, `text/tab-separated-values`), no custom mimetypes registered. The Standard adapter is never reached with non-string content in practice. Keeping the `@var` annotation as-is for now.

File: tests/ProcessorTest.php [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+        self::assertSame('{"id":1}', (string) $psr7Response1->getBody());
+
+        // Second call without map() — mapping from first call still applies (stale mapping)
+        $answer2 = $this->createAnswer();
+        $answer2->setBody(['id' => 2, 'name' => 'Bob', 'email' => 'bob@example.com']);
+
+        $psr7Response2 = $processor->process($answer2, new Response());
+        // Stale mapping is still active — only 'id' key is returned
+        self::assertSame('{"id":2}', (string) $psr7Response2->getBody());
+    }
```
[the remark this answers is not in the document]
Comment (gnutix) [reply tests/ProcessorTest.php]: Fixed: `processor()` is now a factory, test renamed to `testProcessorIsFactoryAndMappingDoesNotLeak`, and assertions updated to verify fresh-instance behavior.

File: lib/Container.php [hunk 3]
```diff
@@ -1,70 +1,83 @@
...
+            ],
+            'codeTranslator' => [],
+            'processorMapping' => [
+                'propertyArrayAccessCheck' => true,
+            ],
+        ], $config);
+
+        $this['response'] = $this->factory(fn (): Response => new Response($this));
+        $this['renderer'] = $this->factory(fn (): Renderer => new Renderer($this));
+        $this['processor'] = fn (): Processor => new Processor($this);
```
[the remark this answers is not in the document]
Comment (gnutix) [reply lib/Container.php]: Fixed: `processor()` now uses `$this->factory()`. Verified against tipee-cow (195 calls) — all usage is single-chain `->map([...])->process(...)`, no code depends on shared state.

File: README.md [hunk 4]
```diff
@@ -1,16 +1,18 @@
 Voilab REST Answer
 ==========
 
-Learn more about it in its [documentation](https://bitbucket.org/voilab/restanswer/src/master/README.md).
+Response system for REST Apis. Provide standard ways to handle rest responses.
```
[the remark this answers is not in the document]
Comment (gnutix) [reply README.md:4]: Won't fix — the README content predates this PR and isn't part of the scope.

File: lib/Processor.php [hunk 5]
```diff
@@ -94,129 +69,96 @@ public function getProcessedResponse(Returnable $returnable) {
...
         if ($isCollection) {
-            // Mapping d'une collection, traite les éléments un par un.
-            if (is_array($content) || $content instanceof \Traversable) {
-                foreach ( $content as &$value ) {
-                    $mapped[] = $this->recursiveMap( $value, $mapping );
+            if (is_iterable($content)) {
+                foreach ($content as $value) {
+                    $mapped[] = $this->recursiveMap($value, $fieldMapping);
                 }
             }
```
[the remark this answers is not in the document]
Comment (gnutix) [reply lib/Processor.php:97]: False positive. ECS is configured with `NativeFunctionInvocationFixer` using default `include` (`@compiler_optimized`). `is_iterable` is not in that set. `ecs check` passes clean.

File: lib/Renderer.php [hunk 6]
```diff
@@ -1,254 +1,147 @@
...
-     * @return self
-     */
-    public function render()
+    public function convert(string $from, string $to): self
     {
-        $this->prepare();
-        $this->engineRender($this->getResponse()->isInterrupt());
+        $converted = iconv($from, $to, $this->content ?? '');
+        $this->content = false !== $converted ? $converted : $this->content;
+
```
[the remark this answers is not in the document]
Comment (gnutix) [reply lib/Renderer.php:69]: False positive. `NativeFunctionInvocationFixer` defaults to `@compiler_optimized` functions only. `sha1()` and `iconv()` are not in that set. `ecs check` passes clean.

File: lib/Processor.php [hunk 7]
```diff
@@ -94,129 +69,96 @@ public function getProcessedResponse(Returnable $returnable) {
...
-                $object = $object->$accessor();
-            } elseif ($this->container['config']['processorMapping']['propertyArrayAccessCheck'] && isset($object[$accessor])) {
-                // ???
-                $object = $object[$accessor];
-            }else {
+            } elseif (is_object($object) && isset($object->{$part})) {
+                $object = $object->{$part};
+            } elseif (is_object($object) && method_exists($object, $part)) {
+                $object = $object->{$part}();
+            } elseif (
```
[the remark this answers is not in the document]
Comment (gnutix) [reply lib/Processor.php:141]: False positive. Same as above — `explode()` and `method_exists()` are not in `@compiler_optimized`. `ecs check` passes clean.

File: lib/ContentType/Text.php [hunk 8]
```diff
@@ -1,30 +1,37 @@
...
         if (is_string($content)) {
             return $content;
-        } elseif (is_object($content)) {
+        }
+
+        if (is_object($content)) {
             if (method_exists($content, 'toString')) {
+                /** @var string */
                 return $content->toString();
             }
```
[the remark this answers is not in the document]
Comment (gnutix) [reply lib/ContentType/Text.php:26]: False positive. `method_exists()` is not in `@compiler_optimized`. `ecs check` passes clean.
